framework/aster-frame/src/cpu.rs

```diff
@@ -79,25 +178,98 @@ impl<T> CpuLocal<T> {
     /// the CPU-local object is only accessed by the current task or IRQ handler.
     /// As local IRQs are disabled, one should keep the closure as short as possible.
     pub fn borrow_with<U, F: FnOnce(&T) -> U>(this: &Self, f: F) -> U {
```

Comment:
This signature may need to add lifetime annotation for `T` and `this`:
```rust
pub fn borrow_with<'a, U, F: FnOnce(&'a T) -> U>(this: &'a Self, f: F) -> U {
```

The reason I'm saying this is because I encountered an issue while using `cpu_local!` where, when `U` is a reference, the compiler seems unable to automatically infer that the lifetime of `U` is consistent with that of `this`.